common/src/View/SetBrushFaceAttributesTool.cpp

```diff
@@ -47,26 +47,36 @@ namespace TrenchBroom {
         }
 
         bool SetBrushFaceAttributesTool::doMouseClick(const InputState& inputState) {
-            return performCopy(inputState, false);
+            if (canPerformCopy(inputState)) {
+                performCopy(inputState, false);
+                return true;
+            } else {
+                return false;
+            }
         }
 
         bool SetBrushFaceAttributesTool::doMouseDoubleClick(const InputState& inputState) {
-            return performCopy(inputState, true);
-        }
-
-        bool SetBrushFaceAttributesTool::performCopy(const InputState& inputState, const bool applyToBrush) {
-            if (!applies(inputState))
+            if (canPerformCopy(inputState)) {
+                // A double click is always preceeded by a single click, so we already done some work which is now
+                // superseded by what is done next. To avoid inconsistencies with undo, we undo the work done by the
+                // single click now:
+                auto document = kdl::mem_lock(m_document);
+                document->undoCommand();
```

Comment:
I'd suggest an added check that the command we're undoing is actually the kind expected (a "performCopy" command applying to this single face) before we undo it. I guess it might be awkward to check that so it's not a huge deal.

Reply:
There is currently no way to check this here I'm afraid. I'm not sure how to do it either, we'd have to add a way to get the last non-transaction command somehow... awkward indeed.